### ISSUE TITLE
[Firmware Request] Unified Bed Leveling

### ISSUE THREAD

**Issue by lin-ycv** [CrealityOfficial/Ender-3S1#22]
I own multiple Ender 3 S1 Pro, I've noticed that on some machines ABL works fine, others not so great, any chance the bed leveling menu can include UBL ability and view mesh ability? something like jyers firmware?
![image](https://user-images.githubusercontent.com/60565737/199474836-5fb6964e-e5b4-4ee2-aaee-d8ef75686cc3.png)

**Comment by ayushsharma82** [comment]
I would have added it myself in the screen if the source code was made available.

**Comment by wizardkitty** [comment]
UBL would be good to have overall but due to creality making some stuff closed source I cannot make a working firmware version that would be able to work, but if creality open sourced some stuff so where we can edit features of the ui we would be good to go.

**Comment by jplmmwolters** [comment]
Yes, would like to have a open source firmware for de LCD screen. @CrealityTech @Creality-mandy

**Comment by wizardkitty** [comment]
This could now be possible. Source Code released for the s1 pro and plus. 

Base implmentation would most likely break some stuff with the touch screen for abl but if possible a upgrade could occur for the S1 Pro and plus with ABL and so on.
